pkg/config/config_test.go

```diff
@@ -3,14 +3,123 @@ package config
 import (
 	"os"
 	"path/filepath"
+	"strings"
 	"testing"
 
+	log "github.com/charmbracelet/log"
 	"github.com/cloudposse/atmos/pkg/schema"
+	"github.com/spf13/cobra"
 	"github.com/spf13/viper"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
+func TestAddConfig_AllTypes(t *testing.T) {
+	tests := []struct {
+		name         string
+		key          string
+		defaultValue interface{}
+		flagValue    string
+		envVar       string
+		envValue     string
+		expected     interface{}
+	}{
+		{
+			name:         "string flag",
+			key:          "app.name",
+			defaultValue: "default-app",
+			flagValue:    "cli-app",
+			envVar:       "APP_NAME_ENV",
+			envValue:     "env-app",
+			expected:     "cli-app",
+		},
```

Comment:
This in the examples below our hard for me to understand because they don't seem to follow any pre-existing Atmos configuration convention.

Reply:
Yes, lets try that lest we confuse the next gen AI that comes to generate code =)

Reply:
Fixed